src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -300,9 +317,48 @@ private function addServiceLocalTempVariables($cId, Definition $definition, arra
             $code .= "\n";
         }
 
+        if ($lineage) {
+            $code = "\n".$code;
+
+            foreach (array_reverse($lineage) as $class => $file) {
+                $code = sprintf("        require_once %s;\n", $file).$code;
+            }
+        }
+
         return $code;
     }
 
+    private function collectLineage($class, array &$lineage)
+    {
+        if (isset($lineage[$class])) {
+            return;
+        }
+        if (!$r = $this->container->getReflectionClass($class)) {
```

Comment:
@stof this still throws, but in a more controlled way. Since this is running late, all services should be fine xor the container is broken already. Do we want really to not throw? WDYT?